### DISCUSSION TITLE
Newbie: How to get File from Url into JavaScript Reader

### DISCUSSION THREAD

**Discussion by Archie-ndw** [nextgenhealthcare/connect#5039] [Q&A] [answered]
Hi,

Do you have any tips or hints for me how to get a File from a Webserver like http://192.168.100.1/dailyfiles/test.csv as Source into a Channel?

I´ve tried to find a solution by google and searching the forums but did not find smth. Would be happy for hints.

I have to fetch a .csv File from an URL as Source for a Channel.
The file is published on a Webserver and can be downloaded with a browser without any Problems.
The URL is like http://192.168.100.1/dailyfiles/test.csv
As this is not a File Sytem and not an FTP or something like that the Standard Source Readers doesnt hlep me.
So my idea was to get the file by JS using the JavaScript Reader
I am used to JS in HTML but did not find a way to get the File since an API like "fetch()" or XMLHttpRequest are not available in mirth.
 I do not have a idea how to get the File into the Channel...

Happy for any help
Andrew

**Comment by pacmano1** [comment, marked as answer]
For a new person I would suggest making the API call in a destination, then collecting the response and routing it to a new channel that has a channel reader as the source.

Otherwise you will need to build to http connection request with java,

**Reply by Archie-ndw** [reply]
Thanks a lot! I was so focused on the Source Connector, that i did not think about the HTML Connector on the Destination side. This will solve my demand in a quite easy way with with standard components.

**Comment by odoodo** [comment]
Use something like
```js
var remoteFile = new java.net.URL("http://192.168.100.1/dailyfiles/test.csv");
var csvFile = org.apache.commons.io.IOUtils.toString(remoteFile.openStream(), java.nio.charset.StandardCharsets.UTF_8);
```
but make sure you use the right encoding for your file (2nd parameter of toString in Apache IOUtils). A list of standard encodings can be found [here](https://docs.oracle.com/javase/8/docs/api/index.html?java/nio/charset/StandardCharsets.html).

**Reply by pacmano1** [reply]
Just a note that Odo's suggestion works if there is no auth required or no token, etc. That the other side is just a "dumb" web server hosting a file.   Also note you are expect to return an xml list from the source tab.

**Reply by Archie-ndw** [reply]
I also tried this way.
Alongside you´ve learned me how to include java classes, so this a worthy learning for me for future projects.
Thanks a lot!
